projects/kit/components/select/native-select/native-select.component.ts

```diff
@@ -0,0 +1,93 @@
+import {DOCUMENT, NgForOf, NgIf, NgTemplateOutlet} from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    computed,
+    inject,
+    Input,
+    signal,
+} from '@angular/core';
+import {tuiAsControl, TuiControl} from '@taiga-ui/cdk/classes';
+import type {TuiBooleanHandler} from '@taiga-ui/cdk/types';
+import {tuiInjectElement} from '@taiga-ui/cdk/utils/dom';
+import {tuiIsPresent} from '@taiga-ui/cdk/utils/miscellaneous';
+import type {TuiTextfieldAccessor} from '@taiga-ui/core/components/textfield';
+import {
+    tuiAsTextfieldAccessor,
+    TuiWithTextfield,
+} from '@taiga-ui/core/components/textfield';
+import {TuiItemsHandlersDirective} from '@taiga-ui/core/directives/items-handlers';
+import {tuiIsFlat} from '@taiga-ui/kit/utils';
+
+@Component({
+    standalone: true,
+    selector: 'select[tuiSelect]',
+    imports: [NgForOf, NgIf, NgTemplateOutlet],
+    templateUrl: './native-select.template.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [tuiAsTextfieldAccessor(TuiNativeSelect), tuiAsControl(TuiNativeSelect)],
+    hostDirectives: [TuiWithTextfield],
+    host: {
+        '[attr.aria-label]': 'ariaLabel',
+        '[attr.aria-invalid]': 'invalid()',
+        '[disabled]': '!interactive()',
+        '[value]': 'stringified()',
+        '(change)': 'selectOption($event.target.options.selectedIndex)',
+    },
+})
+export class TuiNativeSelect<T>
+    extends TuiControl<T | null>
+    implements TuiTextfieldAccessor<T>
+{
+    private readonly doc = inject(DOCUMENT);
+    private readonly element = tuiInjectElement<HTMLSelectElement>();
+
+    protected readonly isFlat = tuiIsFlat;
+    protected readonly placeholder = signal('');
+    protected readonly itemsHandlers = inject(TuiItemsHandlersDirective<T>);
+
+    protected readonly stringified = computed((value = this.value()) =>
+        value ? this.itemsHandlers.stringify()(value) : '',
+    );
+
+    protected readonly showPlaceholder = computed(
+        () => this.placeholder() && !this.stringified(),
+    );
+
+    protected readonly isSelected = computed(
+        (value = this.value()) =>
+            (x: T) =>
+                tuiIsPresent(value) && this.itemsHandlers.identityMatcher()(x, value),
+    );
+
+    @Input()
+    public items: ReadonlyArray<readonly T[]> | readonly T[] | null = [];
+
+    @Input()
+    public labels: readonly string[] = [];
+
+    @Input()
+    public disabledItemHandler!: TuiBooleanHandler<T> | null;
```

Comment:
Why do we need it and not the itemHandlers directive version?

Reply:
https://github.com/taiga-family/taiga-ui/pull/10642/commits/b0c2c153b8dfd9e9952d032cf19798055fe2ebe3